tests/integration/test_interaction.py

```diff
@@ -170,27 +190,51 @@ def parameterized_item_positive_feedback_on_good_move(self, items_map, scroll_do
-            self.wait_until_html_in(definition.feedback_positive, feedback_popup_content)
-            self.assertEqual(popup.get_attribute('class'), 'popup')
-            self.assert_placed_item(definition.item_id, definition.zone_title)
-
-    def parameterized_item_negative_feedback_on_bad_move(self, items_map, all_zones, scroll_down=100, action_key=None):
+            self.wait_until_ondrop_xhr_finished(self._get_item_by_value(definition.item_id))
+            self.assert_placed_item(definition.item_id, definition.zone_title, assessment_mode=assessment_mode)
+            feedback_popup_html = feedback_popup_content.get_attribute('innerHTML')
+            if assessment_mode:
+                self.assertEqual(feedback_popup_html, '')
+                self.assertFalse(popup.is_displayed())
+            else:
+                self.assertEqual(feedback_popup_html, definition.feedback_positive)
+                self.assertEqual(popup.get_attribute('class'), 'popup')
+                self.assertTrue(popup.is_displayed())
+
+    def parameterized_item_negative_feedback_on_bad_move(
+            self, items_map, all_zones, scroll_down=100, action_key=None, assessment_mode=False
+    ):
         popup = self._get_popup()
         feedback_popup_content = self._get_popup_content()
 
         # Scroll drop zones into view to make sure Selenium can successfully drop items
         self.scroll_down(pixels=scroll_down)
 
         for definition in items_map.values():
-            for zone in all_zones:
-                if zone == definition.zone_id:
-                    continue
-                self.place_item(definition.item_id, zone, action_key)
+            # Get first zone that is not correct for this item.
+            zone_id = None
+            zone_title = None
+            for z_id, z_title in all_zones:
+                if z_id != definition.zone_id:
+                    zone_id = z_id
+                    zone_title = z_title
+                    break
+            self.place_item(definition.item_id, zone_id, action_key)
+            if assessment_mode:
+                self.wait_until_ondrop_xhr_finished(self._get_item_by_value(definition.item_id))
+                feedback_popup_html = feedback_popup_content.get_attribute('innerHTML')
+                self.assertEqual(feedback_popup_html, '')
+                self.assertFalse(popup.is_displayed())
+                self.assert_placed_item(definition.item_id, zone_title, assessment_mode=True)
+            else:
                 self.wait_until_html_in(definition.feedback_negative, feedback_popup_content)
                 self.assertEqual(popup.get_attribute('class'), 'popup popup-incorrect')
+                self.assertTrue(popup.is_displayed())
                 self.assert_reverted_item(definition.item_id)
 
-    def parameterized_final_feedback_and_reset(self, items_map, feedback, scroll_down=100, action_key=None):
+    def parameterized_final_feedback_and_reset(
```

Comment:
@mtyaka For this test it might make sense to place *all* items on the board when in assessment mode (to verify that incorrectly-placed items get returned to the item bank as well).